src/main/webapp/js/index.js

```diff
@@ -12,8 +12,94 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-/* Date range picker handler:
- * https://www.daterangepicker.com/
+
+/** Fetch receipts from the server and add them to the DOM. */
+async function searchReceipts() {
+  const label = document.getElementById('search-input').value;
+  const response = await fetch(`/search-receipts?label=${label}`);
+  const receipts = await response.json();
+
+  clearExistingReceipts();
+  displayReceipts(label, receipts);
+}
+
+/**
+ * Clear out existing receipts display.
+ *
+ * Note: 3 childNodes make up a receipt card.
+ * The first 3 nodes are the template receipt card and each subsequent
+ * 3 are clones of the template. The template must never be deleted or
+ * it'll be unusable in the future.
```

Comment:
Could the template be moved outside of receipts-display so you don't have to worry about accidentally deleting the template?

Reply:
No, receipts-display is the container that allows the inner receipt cards to be responsive :/

Reply:
I think this is moot now that you are using the querySelector, but the template itself I think can be placed anywhere in the document, but once you do clone the template, it does have to be appended as a child of receipts-display